elastic/kibana
# [Response Ops][Alerting] Delete associated backfill jobs when a rule is deleted

Resolves https://github.com/elastic/kibana/issues/182240

## Summary

Queries for and deletes any pending backfill tasks for a rule when a rule is deleted.

## To Verify

1. Create a detection rule
2. Schedule a backfill for the rule with a long time interval that will take a while to run

```
POST /internal/alerting/rules/backfill/_schedule
[
  {
    "rule_id": <ruleId>,
    "start": "2024-05-01T12:00:00.000Z",
    "end": "2024-05-10T12:00:00.000Z",
  }
]
```

3. Make sure you can retrieve the backfill info: `GET kbn:/internal/alerting/rules/backfill/{backfillId}`
4. Delete the rule via the UI or API
5. Try to get the backfill info again. The job should be deleted
